### PR TITLE
[MBL-1391] Payment Intent StripeIntentContextType and CheckoutID

### PR DESCRIPTION
**Pull request by scottkicks**
# 📲 What

Updates the call to CreatePaymentIntent to include the StripeIntentContextType and the Checkout ID. 
- We added the context somewhat recently for the SetupIntent calls. Same idea here.

# 🤔 Why

We need this information to store additional metadata in the payment intent when it's created for on-session payments. This is helpful when the backend tries to debug any issues during checkout.

# 🛠 How

**Checkout ID**
- Update GraphAPI to expose the new property
- Update CreatePaymentIntentInput
- Pass the id returned from `CreateCheckoutMutation` to .createPaymentIntent call sites
   - This is already available to us. Just need to pass it around accordingly
- Make sure to encode the id so the backend receives it in base 64.

**Context**
- Adds `paymentIntentContext` (defined in GraphAPI model) to `CreatePaymentIntentInput`
- Updates `.createPaymentIntent` call sites

# 👀 See

No UI changes

# ♿️ Accessibility 

No accessibility changes needed

# 🏎 Performance

Same

# ✅ Acceptance criteria

- [x] No changes to normal Crowdfunding checkout flow
- [x] No changes to Late Pledge checkout flow
- [x] No changes to creating Setup Intents and Payment Intents

# ⏰ Next Steps
- [ ] Update the Late pledge feature flag to block late pledges on app versions that don't pass in the additional metadata (i.e. older app versions)
- [x] Note this for regression testing in the upcoming release since this touches pieces of both checkout flows.

### CODE REVIEW

File: KsApi/mutations/inputs/CreatePaymentIntentInput.swift [hunk 1]
```diff
@@ -4,21 +4,25 @@ public struct CreatePaymentIntentInput: GraphMutationInput, Encodable {
...
-    digitalMarketingAttributed: Bool?
+    digitalMarketingAttributed: Bool?,
+    paymentIntentContext: GraphAPI.StripeIntentContextTypes?
   ) {
     self.projectId = projectId
     self.amountDollars = amountDollars
     self.digitalMarketingAttributed = digitalMarketingAttributed
+    self.paymentIntentContext = paymentIntentContext
   }
 }
```
Comment (amy-at-kickstarter) [inline KsApi/mutations/inputs/CreatePaymentIntentInput.swift:28]: Out of scope for this PR, but this type seems redundant to `GraphAPI.CreatePaymentIntentInput`

**Review by amy-at-kickstarter — APPROVED** [review]
LGTM

File: KsApi/mutations/inputs/CreatePaymentIntentInput.swift [hunk 2]
```diff
@@ -4,21 +4,25 @@ public struct CreatePaymentIntentInput: GraphMutationInput, Encodable {
...
     projectId: String,
     amountDollars: String,
-    digitalMarketingAttributed: Bool?
+    digitalMarketingAttributed: Bool?,
+    paymentIntentContext: GraphAPI.StripeIntentContextTypes?
   ) {
     self.projectId = projectId
     self.amountDollars = amountDollars
     self.digitalMarketingAttributed = digitalMarketingAttributed
+    self.paymentIntentContext = paymentIntentContext
```
Comment (amy-at-kickstarter) [inline KsApi/mutations/inputs/CreatePaymentIntentInput.swift:26]: Is this ever _not_ `.postCampaignCheckout`? i.e. do we ever use a payment intent for a live pledge? You could potentially just hard-code it here.

Comment (scottkicks) [reply KsApi/mutations/inputs/CreatePaymentIntentInput.swift]: That's a good point! I doubt we'll need a payment intent in a different context unless, for some weird reason, we want to create one from the account settings -> payment methods screen.

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Library/ViewModels/PledgePaymentMethodsViewModel.swift [hunk 3]
```diff
@@ -334,6 +337,7 @@ public final class PledgePaymentMethodsViewModel: PledgePaymentMethodsViewModelT
         )
         clientSecretSignal = stripeIntentService.createPaymentIntent(
           for: project.graphID,
+          checkoutId: checkoutId,
```
Comment (amy-at-kickstarter) [inline Library/ViewModels/PledgePaymentMethodsViewModel.swift:340]: Is this one already graph-encoded?

Comment (scottkicks) [reply Library/ViewModels/PledgePaymentMethodsViewModel.swift:340]: Nope!

**Pull request merged**